ddl/index.go

```diff
@@ -61,7 +61,7 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.GenWithStack("column does not exist: %s", ic.Column.Name)
 		}
 
-		if col.Flen == 0 {
+		if col.Flen == 0 && (types.IsTypeChar(col.FieldType.Tp) || types.IsTypeVarchar(col.FieldType.Tp)) {
```

Comment:
could we use `IsString`? I'm not sure the type blob should check the length or not.

Reply:
@jackysp  `IsTypeBlob` is different errors, so I only use these two.

Reply:
OK.